File: lc_soln/1695_maximum_erasure_value/solution.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        unordered_set<int> seen;
        int curr = 0;
        int result = 0;
        int start = 0;
        for (int i = 0; i < nums.size(); ++i) {
            int num = nums[i];
            while (seen.contains(num)) {
                seen.erase(nums[start]);
                curr -= nums[start];
                ++start;
            }
            seen.insert(num);
            curr += num;
            result = max(result, curr);
        }
        return result;
    }
};

File: lc_soln/1695_maximum_erasure_value/solution.rs
use std::collections::HashSet;

struct Solution;

impl Solution {
    pub fn maximum_unique_subarray(nums: Vec<i32>) -> i32 {
        let mut seen: HashSet<i32> = HashSet::new();
        let mut curr = 0;
        let mut result = 0;
        let mut start = 0;
        for num in &nums {
            while seen.contains(num) {
                seen.remove(&nums[start]);
                curr -= nums[start];
                start += 1;
            }
            seen.insert(*num);
            curr += num;
            result = result.max(curr);
        }
        result
    }
}

fn main() {
    println!("{}", Solution::maximum_unique_subarray(vec![4, 2, 4, 5, 6]));
}
